src/Http/Http.Extensions/src/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Net.Http.Headers;
+
+#nullable enable
+
+namespace Microsoft.AspNetCore.Http
+{
+    public static class HttpRequestExtensions
+    {
+        public static bool HasJsonContentType(this HttpRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            if (!MediaTypeHeaderValue.TryParse(request.ContentType, out var mt))
```

Comment:
Ugh, allocations...

Reply:
Agree. It is simple enough to improve this when there is a better API.

Reply:
cc @Tratcher